aten/src/ATen/native/mkl/SparseBlasImpl.cpp

```diff
@@ -600,12 +608,30 @@ void triangular_solve_out_sparse_csr(
       "Calling triangular_solve on a sparse CPU tensor requires Linux platform. ",
       "Please use PyTorch built with MKL on Linux.");
 #else
-  if (B.numel() == 0 || X.numel() == 0 || A._nnz() == 0) {
+  if (B.numel() == 0 || X.numel() == 0 || A_._nnz() == 0) {
     // If A has no nnz, then A is singular and we can't solve.
     X.fill_(NAN);
     return;
   }
 
+  const auto materialize_diagonal_indices = [](const Tensor& t) -> Tensor {
+    const auto n = t.size(-1);
+    const auto diag_indices = at::arange(n, t.crow_indices().options()).unsqueeze(0).expand({2, n});
+    const auto diag_values = at::zeros(n, t.values().options());
```

Comment:
Let's wait until the CIs results, but it seems like we can replace `zeros` with `empty`.